homeassistant/components/hyperion/sensor.py

```diff
@@ -0,0 +1,189 @@
+) -> None:
+    """Set up a Hyperion platform from config entry."""
+    entry_data = hass.data[DOMAIN][config_entry.entry_id]
+    server_id = config_entry.unique_id
+
+    @callback
+    def instance_add(instance_num: int, instance_name: str) -> None:
+        """Add entities for a new Hyperion instance."""
+        assert server_id
+        sensors = []
+        sensors.append(
+            HyperionSensor(
+                server_id,
+                instance_num,
+                instance_name,
+                entry_data[CONF_INSTANCE_CLIENTS][instance_num],
+                SensorEntityDescription(
+                    key="effect",
+                    translation_key="effect",
+                    icon="mdi:lava-lamp",
+                    entity_registry_enabled_default=True,
+                )                
+            ),
+        )
+
+        async_add_entities(sensors)
+
+    @callback
+    def instance_remove(instance_num: int) -> None:
+        """Remove entities for an old Hyperion instance."""
+        assert server_id
+        async_dispatcher_send(
+            hass,
+            SIGNAL_ENTITY_REMOVE.format(
+                _sensor_unique_id(server_id, instance_num),
+            ),
+        )
+
+    listen_for_instance_updates(hass, config_entry, instance_add, instance_remove)
+
+
+class HyperionSensor(SensorEntity):
+    """Sensor class."""
+
+    _attr_has_entity_name = True
+    _attr_should_poll = False
+
+    def __init__(
+        self,
+        server_id: str,
+        instance_num: int,
+        instance_name: str,
+        hyperion_client: client.HyperionClient,
+        entity_description: SensorEntityDescription | None = None,
+    ) -> None:
+        """Initialize the sensor."""
+        self._attr_unique_id = _sensor_unique_id(
+            server_id, instance_num
+        )
+
+        if entity_description:
+            self.entity_description = entity_description
+        elif not hasattr(self, "entity_description"):
+            raise ValueError(
+                '"entity_description" must be either set as class variable or passed on init!'
+            )
+                
+        self._device_id = get_hyperion_device_id(server_id, instance_num)
+        self._instance_name = instance_name
+        self._client = hyperion_client
+        self._client_callbacks = {
+            f"{KEY_PRIORITIES}-{KEY_UPDATE}": self._update_priorities
+        }
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self._device_id)},
+            manufacturer=HYPERION_MANUFACTURER_NAME,
+            model=HYPERION_MODEL_NAME,
+            name=self._instance_name,
+            configuration_url=self._client.remote_url,
+        )
+
+
+    @property
+    def state(self):
+        """Return the active priority's effect."""
+        for priority in self._client.priorities or []:
+            if priority[KEY_VISIBLE] == True:
+                return priority[KEY_OWNER]
+
+        return "unknown"
```

Comment:
This would return `unknown` for a color. How about you return the RGB value in that case?